From 59b898e02f7c3e2c223b859848b94e26c8b0ff80 Mon Sep 17 00:00:00 2001
From: Valentin David <valentin.david@codethink.co.uk>
Date: Mon, 3 Feb 2020 14:37:18 +0100
Subject: [PATCH 3/4] Add Bluez

This is a build dependency for PipeWire. PipeWire requires a bit more
than what
`components/bluez-headers.bst`. `components/bluez-headers.bst` remains
in order to avoid a circular dependency with `components/python3.bst`.
---
 elements/components/bluez-headers.bst | 11 ++++++-----
 elements/components/bluez.bst         | 24 ++++++++++++++++++++++++
 elements/include/bluez.yml            |  5 +++++
 3 files changed, 35 insertions(+), 5 deletions(-)
 create mode 100644 elements/components/bluez.bst
 create mode 100644 elements/include/bluez.yml

diff --git a/elements/components/bluez-headers.bst b/elements/components/bluez-headers.bst
index 331e6216..5ac38b13 100644
--- a/elements/components/bluez-headers.bst
+++ b/elements/components/bluez-headers.bst
@@ -1,4 +1,8 @@
 kind: manual
+description: |
+  This element provides just the headers required to build
+  Python. Because Bluez depends on Python, we cannot build Python without
+  this element. Full elements is `components/bluez.bst`.
 
 build-depends:
 - bootstrap-import.bst
@@ -8,8 +12,5 @@ config:
   - |
     install -D -m644 -t "%{install-root}%{includedir}/bluetooth" lib/*.h
 
-sources:
-- kind: git_tag
-  url: kernel:bluetooth/bluez.git
-  track: master
-  ref: 5.51-0-g6de4bdb957cdc85d89851420ab06ca8e226f8d4e
+(@): elements/include/bluez.yml
+
diff --git a/elements/components/bluez.bst b/elements/components/bluez.bst
new file mode 100644
index 00000000..55ec909b
--- /dev/null
+++ b/elements/components/bluez.bst
@@ -0,0 +1,24 @@
+kind: autotools
+
+depends:
+- bootstrap-import.bst
+- components/glib.bst
+- components/dbus.bst
+- components/libical.bst
+
+build-depends:
+- public-stacks/buildsystem-autotools.bst
+- components/systemd.bst
+
+variables:
+  conf-local: >-
+    --enable-library
+
+public:
+  bst:
+    split-rules:
+      devel:
+        (>):
+        - '%{libdir}/libbluetooth.so'
+
+(@): elements/include/bluez.yml
diff --git a/elements/include/bluez.yml b/elements/include/bluez.yml
new file mode 100644
index 00000000..1efd068d
--- /dev/null
+++ b/elements/include/bluez.yml
@@ -0,0 +1,5 @@
+sources:
+- kind: git_tag
+  url: kernel:bluetooth/bluez.git
+  track: master
+  ref: 5.52-0-ge002fd43636090ac246f80f5de6195055384dfed
-- 
2.26.2

